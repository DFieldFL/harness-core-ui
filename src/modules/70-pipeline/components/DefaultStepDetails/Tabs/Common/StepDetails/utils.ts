export function encodeURIWithReservedChars(uri: string): string {
  return encodeURIComponent(uri).replace(/[;,/?:@&=+$#]/g, function (c) {
    return '%' + c.charCodeAt(0).toString(16)
  })
}
