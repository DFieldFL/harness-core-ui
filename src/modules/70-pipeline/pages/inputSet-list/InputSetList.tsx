import React, { useEffect, useState } from 'react'
import { isEmpty } from 'lodash-es'
import { Popover, Layout, TextInput, useModalHook, Text, Color, ButtonVariation } from '@wings-software/uicore'
import { Menu, MenuItem, Position } from '@blueprintjs/core'
import { useHistory, useParams } from 'react-router-dom'
import { Page } from '@common/exports'
import {
  InputSetSummaryResponse,
  useGetInputSetsListForPipeline,
  useGetTemplateFromPipeline
} from 'services/pipeline-ng'
import { OverlayInputSetForm } from '@pipeline/components/OverlayInputSetForm/OverlayInputSetForm'
import routes from '@common/RouteDefinitions'
import type { GitQueryParams, PipelinePathProps, PipelineType } from '@common/interfaces/RouteInterfaces'
import { useDocumentTitle } from '@common/hooks/useDocumentTitle'
import { useStrings } from 'framework/strings'
import RbacButton from '@rbac/components/Button/Button'
import { PermissionIdentifier } from '@rbac/interfaces/PermissionIdentifier'
import { ResourceType } from '@rbac/interfaces/ResourceType'
import { usePermission } from '@rbac/hooks/usePermission'
import { useQueryParams } from '@common/hooks'
import { InputSetListView } from './InputSetListView'
import css from './InputSetList.module.scss'

const InputSetList: React.FC = (): JSX.Element => {
  const [searchParam, setSearchParam] = React.useState('')
  const [page, setPage] = React.useState(0)
  const { repoIdentifier, branch } = useQueryParams<GitQueryParams>()
  const { projectIdentifier, orgIdentifier, accountId, pipelineIdentifier, module } = useParams<
    PipelineType<PipelinePathProps> & { accountId: string }
  >()

  const {
    data: inputSet,
    loading,
    refetch,
    error
  } = useGetInputSetsListForPipeline({
    queryParams: {
      accountIdentifier: accountId,
      orgIdentifier,
      projectIdentifier,
      pipelineIdentifier,
      pageIndex: page,
      pageSize: 10,
      searchTerm: searchParam,
      ...(!isEmpty(repoIdentifier) && !isEmpty(branch)
        ? {
            repoIdentifier,
            branch,
            getDefaultFromOtherRepo: true
          }
        : {})
    },
    debounce: 300
  })

  const { data: template } = useGetTemplateFromPipeline({
    queryParams: {
      accountIdentifier: accountId,
      orgIdentifier,
      pipelineIdentifier,
      projectIdentifier,
      repoIdentifier,
      branch
    }
  })

  // These flags will be used to disable the Add Input set buttons in the page.
  const [pipelineHasRuntimeInputs, setPipelineHasRuntimeInputs] = useState(true)
  useEffect(() => {
    if (!template?.data?.inputSetTemplateYaml) {
      setPipelineHasRuntimeInputs(false)
    } else {
      setPipelineHasRuntimeInputs(true)
    }
  }, [template])

  const [selectedInputSet, setSelectedInputSet] = React.useState<{
    identifier?: string
    repoIdentifier?: string
    branch?: string
  }>()
  const history = useHistory()
  const { getString } = useStrings()
  useDocumentTitle([getString('pipelines'), getString('inputSetsText')])

  const goToInputSetForm = React.useCallback(
    (inputSetTemp?: InputSetSummaryResponse) => {
      history.push(
        routes.toInputSetForm({
          accountId,
          orgIdentifier,
          projectIdentifier,
          pipelineIdentifier,
          inputSetIdentifier: typeof inputSetTemp?.identifier !== 'string' ? '-1' : inputSetTemp.identifier,
          module,
          inputSetRepoIdentifier: inputSetTemp?.gitDetails?.repoIdentifier,
          inputSetBranch: inputSetTemp?.gitDetails?.branch,
          repoIdentifier,
          branch
        })
      )
    },
    [accountId, orgIdentifier, projectIdentifier, pipelineIdentifier, module, history, repoIdentifier, branch]
  )

  const [canUpdateInputSet] = usePermission(
    {
      resourceScope: {
        accountIdentifier: accountId,
        orgIdentifier,
        projectIdentifier
      },
      resource: {
        resourceType: ResourceType.PIPELINE,
        resourceIdentifier: pipelineIdentifier
      },
      permissions: [PermissionIdentifier.EDIT_PIPELINE]
    },
    [accountId, orgIdentifier, projectIdentifier, pipelineIdentifier]
  )

  const [showOverlayInputSetForm, hideOverlayInputSetForm] = useModalHook(
    () => (
      <OverlayInputSetForm
        identifier={selectedInputSet?.identifier}
        overlayInputSetRepoIdentifier={selectedInputSet?.repoIdentifier}
        overlayInputSetBranch={selectedInputSet?.branch}
        hideForm={() => {
          refetch()
          hideOverlayInputSetForm()
        }}
        isReadOnly={!canUpdateInputSet}
      />
    ),
    [selectedInputSet]
  )

  return (
    <>
      <Page.SubHeader>
        <Layout.Horizontal>
          <Popover
            minimal
            content={
              <Menu className={css.menuList}>
                <MenuItem
                  text={getString('inputSets.inputSetLabel')}
                  onClick={() => {
                    goToInputSetForm()
                  }}
                />
                <MenuItem
                  text={getString('inputSets.overlayInputSet')}
                  onClick={() => {
                    setSelectedInputSet({ identifier: '', repoIdentifier, branch })
                    showOverlayInputSetForm()
                  }}
                />
              </Menu>
            }
            position={Position.BOTTOM}
            disabled={!canUpdateInputSet || !pipelineHasRuntimeInputs}
          >
            <RbacButton
              text={getString('inputSets.newInputSet')}
              rightIcon="caret-down"
              variation={ButtonVariation.PRIMARY}
              permission={{
                resource: {
                  resourceType: ResourceType.PIPELINE,
                  resourceIdentifier: pipelineIdentifier
                },
                permission: PermissionIdentifier.EDIT_PIPELINE
              }}
              disabled={!pipelineHasRuntimeInputs}
              tooltip={
                !pipelineHasRuntimeInputs ? (
                  <Text padding="medium">{getString('pipeline.inputSets.noRuntimeInputsCurrently')}</Text>
                ) : undefined
              }
            />
          </Popover>
        </Layout.Horizontal>

        <Layout.Horizontal spacing="small">
          <TextInput
            leftIcon={'thinner-search'}
            leftIconProps={{ name: 'thinner-search', size: 14, color: Color.GREY_700 }}
            placeholder={getString('inputSets.searchInputSet')}
            wrapperClassName={css.searchWrapper}
            value={searchParam}
            onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
              setSearchParam(e.target.value.trim())
            }}
          />
        </Layout.Horizontal>
      </Page.SubHeader>

      <Page.Body
        loading={loading}
        error={error?.message}
        retryOnError={/* istanbul ignore next */ () => refetch()}
        noData={{
          when: () => !inputSet?.data?.content?.length,
          icon: 'yaml-builder-input-sets',
          message: getString('inputSets.aboutInputSets'),
          buttonText: getString('inputSets.addInputSet'),
          onClick: () => goToInputSetForm(),
          buttonDisabled: !canUpdateInputSet || !pipelineHasRuntimeInputs,
          buttonDisabledTooltip: getString('pipeline.inputSets.noRuntimeInputsCurrently')
        }}
      >
        <InputSetListView
          data={inputSet?.data}
          gotoPage={setPage}
          pipelineHasRuntimeInputs={pipelineHasRuntimeInputs}
          goToInputSetDetail={inputSetTemp => {
            setSelectedInputSet({
              identifier: inputSetTemp?.identifier,
              repoIdentifier: inputSetTemp?.gitDetails?.repoIdentifier,
              branch: inputSetTemp?.gitDetails?.branch
            })
            if (inputSetTemp?.inputSetType === 'INPUT_SET') {
              goToInputSetForm(inputSetTemp)
            } else {
              showOverlayInputSetForm()
            }
          }}
          refetchInputSet={refetch}
          canUpdate={canUpdateInputSet}
        />
      </Page.Body>
    </>
  )
}

export default InputSetList
