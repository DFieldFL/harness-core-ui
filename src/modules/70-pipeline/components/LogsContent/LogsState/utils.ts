import { escapeStringRegexp } from '@common/utils/StringUtils'

import type { State } from './types'

export function getRegexForSearch(str: string): RegExp {
  return new RegExp(escapeStringRegexp(str), 'gi')
}

export function getDefaultReducerState(props: Partial<Pick<State, 'selectedStage' | 'selectedStep'>> = {}): State {
  return {
    units: [],
    logKeys: [],
    selectedStage: props.selectedStage || '',
    selectedStep: props.selectedStep || '',
    dataMap: {},
    searchData: {
      text: '',
      currentIndex: 0,
      linesWithResults: []
    }
  }
}

export const logsCache = new Map<string, string>()
