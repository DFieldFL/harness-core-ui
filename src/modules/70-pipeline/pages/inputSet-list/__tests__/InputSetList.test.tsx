import React from 'react'
import { render, RenderResult, waitFor, fireEvent, getByText, queryByText, getByTestId } from '@testing-library/react'
import { act } from 'react-dom/test-utils'
import { TestWrapper, findDialogContainer, findPopoverContainer } from '@common/utils/testUtils'
import routes from '@common/RouteDefinitions'
import { defaultAppStoreValues } from '@common/utils/DefaultAppStoreData'
import { accountPathProps, pipelineModuleParams, pipelinePathProps } from '@common/utils/routeUtils'
import InputSetList from '../InputSetList'
import {
  TemplateResponse,
  PipelineResponse,
  ConnectorResponse,
  GetInputSetsResponse,
  GetInputSetEdit,
  MergeInputSetResponse,
  GetOverlayInputSetEdit
} from './InputSetListMocks'

jest.mock('@common/components/YAMLBuilder/YamlBuilder', () => ({ children }: { children: JSX.Element }) => (
  <div>{children}</div>
))
const deleteInputSetMock = jest.fn()
const deleteInputSet = (): Promise<{ status: string }> => {
  deleteInputSetMock()
  return Promise.resolve({ status: 'SUCCESS' })
}
const getInputSetList = jest.fn()

jest.mock('services/cd-ng', () => ({
  useGetConnector: jest.fn(() => ConnectorResponse)
}))

jest.mock('services/pipeline-ng', () => ({
  useGetPipeline: jest.fn(() => PipelineResponse),
  useGetTemplateFromPipeline: jest.fn(() => TemplateResponse),
  useGetMergeInputSetFromPipelineTemplateWithListInput: jest.fn(() => MergeInputSetResponse),
  useGetOverlayInputSetForPipeline: jest.fn(() => GetOverlayInputSetEdit),
  useCreateInputSetForPipeline: jest.fn(() => ({})),
  useUpdateInputSetForPipeline: jest.fn(() => ({})),
  useUpdateOverlayInputSetForPipeline: jest.fn(() => ({})),
  useCreateOverlayInputSetForPipeline: jest.fn(() => ({})),
  useGetInputSetsListForPipeline: jest.fn().mockImplementation(args => {
    getInputSetList(args)
    return GetInputSetsResponse
  }),
  getInputSetForPipelinePromise: jest.fn().mockImplementation(() => Promise.resolve(GetInputSetsResponse.data)),
  useGetInputSetForPipeline: jest.fn(() => GetInputSetEdit),
  useDeleteInputSetForPipeline: jest.fn().mockImplementation(() => ({ mutate: deleteInputSet })),
  usePostPipelineExecuteWithInputSetYaml: jest.fn(() => ({})),
  useGetYamlSchema: jest.fn(() => ({}))
}))

const TEST_PATH = routes.toInputSetList({ ...accountPathProps, ...pipelinePathProps, ...pipelineModuleParams })

describe('Input Set List tests', () => {
  test('render Input Set List view', async () => {
    const { getAllByText, container } = render(
      <TestWrapper
        path={TEST_PATH}
        pathParams={{
          accountId: 'testAcc',
          orgIdentifier: 'testOrg',
          projectIdentifier: 'test',
          pipelineIdentifier: 'pipeline',
          module: 'cd'
        }}
        defaultAppStoreValues={defaultAppStoreValues}
      >
        <InputSetList />
      </TestWrapper>
    )
    await waitFor(() => getAllByText('OverLayInput'))
    expect(container).toMatchSnapshot()
  })
})

describe('Input Set List - Actions tests', () => {
  let container: HTMLElement | undefined
  let getAllByText: RenderResult['getAllByText'] | undefined

  beforeEach(async () => {
    const renderObj = render(
      <TestWrapper
        path={TEST_PATH}
        pathParams={{
          accountId: 'testAcc',
          orgIdentifier: 'testOrg',
          projectIdentifier: 'test',
          pipelineIdentifier: 'pipeline',
          module: 'cd'
        }}
        defaultAppStoreValues={defaultAppStoreValues}
      >
        <InputSetList />
      </TestWrapper>
    )
    container = renderObj.container
    getAllByText = renderObj.getAllByText
    await waitFor(() => getAllByText?.('OverLayInput'))
  })

  test('click handler for Input Set', async () => {
    const inputSetRow = getAllByText?.('asd')[0]
    await act(async () => {
      fireEvent.click(inputSetRow!)
      await waitFor(() => getByTestId(document.body, 'location'))
      expect(
        getByTestId(document.body, 'location').innerHTML.endsWith(
          routes.toInputSetForm({
            accountId: 'testAcc',
            orgIdentifier: 'testOrg',
            projectIdentifier: 'test',
            pipelineIdentifier: 'pipeline',
            inputSetIdentifier: 'asd',
            module: 'cd'
          } as any)
        )
      ).toBeTruthy()
    })
  })

  test('click handler for Overlay Input Set', async () => {
    const inputSetRow = getAllByText?.('OverLayInput')[0]
    await act(async () => {
      fireEvent.click(inputSetRow!)
      await waitFor(() => getByText(document.body, 'Edit Overlay Input Set: OverLayInput'))
      const form = findDialogContainer()
      expect(form).toBeTruthy()
    })
  })

  test('click handler for edit Overlay Input Set from menu', async () => {
    const menu = container?.querySelectorAll("[icon='more']")[0]
    fireEvent.click(menu!)
    const editMenu = getAllByText?.('Edit')[0]
    await act(async () => {
      fireEvent.click(editMenu!)
      await waitFor(() => getByText(document.body, 'Edit Overlay Input Set: OverLayInput'))
      const form = findDialogContainer()
      expect(form).toBeTruthy()
    })
  })

  test('click handler for edit Input Set from menu', async () => {
    const menu = container?.querySelectorAll("[icon='more']")[1]
    fireEvent.click(menu!)
    const editMenu = getAllByText?.('Edit')[0]
    await act(async () => {
      fireEvent.click(editMenu!)
      await waitFor(() => getByTestId(document.body, 'location'))
      expect(
        getByTestId(document.body, 'location').innerHTML.endsWith(
          routes.toInputSetForm({
            accountId: 'testAcc',
            orgIdentifier: 'testOrg',
            projectIdentifier: 'test',
            pipelineIdentifier: 'pipeline',
            inputSetIdentifier: 'asd',
            module: 'cd'
          } as any)
        )
      ).toBeTruthy()
    })
  })

  test('click handler for delete Overlay Input Set from menu', async () => {
    deleteInputSetMock.mockReset()
    const menu = container?.querySelectorAll("[icon='more']")[0]
    fireEvent.click(menu!)
    const deleteMenu = getAllByText?.('Delete')[0]
    await act(async () => {
      fireEvent.click(deleteMenu!)
      await waitFor(() => getByText(document.body, 'Delete Overlay Input Set'))
      const form = findDialogContainer()
      expect(form).toBeTruthy()
      const deleteBtn = queryByText(form as HTMLElement, 'Delete')
      fireEvent.click(deleteBtn!)
      await waitFor(() => getByText(document.body, 'Input Set "OverLayInput" deleted'))
      expect(deleteInputSetMock).toBeCalled()
    })
  })

  test('click handler for delete Input Set from menu', async () => {
    deleteInputSetMock.mockReset()
    const menu = container?.querySelectorAll("[icon='more']")[1]
    fireEvent.click(menu!)
    const deleteMenu = getAllByText?.('Delete')[0]
    await act(async () => {
      fireEvent.click(deleteMenu!)
      await waitFor(() => getByText(document.body, 'Delete Input Set'))
      const form = findDialogContainer()
      expect(form).toBeTruthy()
      const deleteBtn = queryByText(form as HTMLElement, 'Delete')
      fireEvent.click(deleteBtn!)
      await waitFor(() => getByText(document.body, 'Input Set "asd" deleted'))
      expect(deleteInputSetMock).toBeCalled()
    })
  })

  test('click handler for new Overlay Input Set', async () => {
    const menu = getAllByText?.('+ New Input Set')[0]
    fireEvent.click(menu!)
    const popover = findPopoverContainer()
    const newInputSet = getByText(popover as HTMLElement, 'Overlay Input Set')
    await act(async () => {
      fireEvent.click(newInputSet)
      await waitFor(() => getByText(document.body, 'New Overlay Input Set'))
      let form = findDialogContainer()
      expect(form).toBeTruthy()
      // Close
      // eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
      fireEvent.click(form?.querySelector('[icon="small-cross"]')!)
      form = findDialogContainer()
      expect(form).not.toBeTruthy()
    })
  })

  test('click handler for new Input Set', async () => {
    const menu = getAllByText?.('+ New Input Set')[0]
    fireEvent.click(menu!)
    const popover = findPopoverContainer()
    const newInputSet = getByText(popover as HTMLElement, 'Input Set')
    await act(async () => {
      fireEvent.click(newInputSet)
      await waitFor(() => getByTestId(document.body, 'location'))
      expect(
        getByTestId(document.body, 'location').innerHTML.endsWith(
          routes.toInputSetForm({
            accountId: 'testAcc',
            orgIdentifier: 'testOrg',
            projectIdentifier: 'test',
            pipelineIdentifier: 'pipeline',
            inputSetIdentifier: '-1',
            module: 'cd'
          } as any)
        )
      ).toBeTruthy()
    })
  })

  test('search Input Set list', async () => {
    getInputSetList.mockReset()
    const searchInput = container?.querySelector('[placeholder="Search Input Set"]') as HTMLInputElement
    fireEvent.change(searchInput, { target: { value: 'asd' } })
    expect(getInputSetList).toBeCalledWith({
      debounce: 300,
      queryParams: {
        accountIdentifier: 'testAcc',
        orgIdentifier: 'testOrg',
        pageIndex: 0,
        pageSize: 10,
        pipelineIdentifier: 'pipeline',
        projectIdentifier: 'test',
        searchTerm: 'asd'
      }
    })
  })
})
