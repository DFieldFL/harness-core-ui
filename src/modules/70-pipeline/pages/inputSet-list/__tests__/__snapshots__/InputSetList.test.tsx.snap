// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Set List tests render Input Set List view 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main container standard StyledProps--flex StyledProps--padding-left-large StyledProps--padding-right-large"
  >
    <span
      class="bp3-popover-wrapper"
    >
      <span
        class="bp3-popover-target"
      >
        <button
          class="bp3-button StyledProps--font Button--button StyledProps--main StyledProps--intent-primary"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            + New Input Set
          </span>
          <span
            class="bp3-icon bp3-icon-caret-down StyledProps--main"
            icon="caret-down"
          >
            <svg
              data-icon="caret-down"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                caret-down
              </desc>
              <path
                d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </button>
      </span>
    </span>
    <div
      class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
    >
      <div
        class="TextInput--main"
      >
        <div
          class="bp3-input-group search"
        >
          <span
            class="bp3-icon bp3-icon-search StyledProps--main"
            icon="search"
          >
            <svg
              data-icon="search"
              height="12"
              viewBox="0 0 16 16"
              width="12"
            >
              <desc>
                search
              </desc>
              <path
                d="M15.55 13.43l-2.67-2.68a6.94 6.94 0 001.11-3.76c0-3.87-3.13-7-7-7s-7 3.13-7 7 3.13 7 7 7c1.39 0 2.68-.42 3.76-1.11l2.68 2.67a1.498 1.498 0 102.12-2.12zm-8.56-1.44c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <input
            class="bp3-input"
            placeholder="Search Input Set"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="StyledProps--font StyledProps--main pageBody"
  >
    <div
      class="table table"
    >
      <div
        class="header"
        role="row"
      >
        <div
          class="cell"
          colspan="1"
          role="columnheader"
          style="width: 25%;"
          title="Toggle SortBy"
        >
          INPUT SET
          <span
            class="bp3-icon bp3-icon-double-caret-vertical StyledProps--main StyledProps--padding-left-small"
            icon="double-caret-vertical"
          >
            <svg
              data-icon="double-caret-vertical"
              height="15"
              viewBox="0 0 16 16"
              width="15"
            >
              <desc>
                double-caret-vertical
              </desc>
              <path
                d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
        <div
          class="cell"
          colspan="1"
          role="columnheader"
          style="width: 20%;"
        >
          DESCRIPTION
        </div>
        <div
          class="cell"
          colspan="1"
          role="columnheader"
          style="width: 10%;"
        >
          INPUT FIELD SUMMARY
        </div>
        <div
          class="cell"
          colspan="1"
          role="columnheader"
          style="width: 10%;"
        >
          LAST UPDATED BY
        </div>
        <div
          class="cell"
          colspan="1"
          role="columnheader"
          style="width: 10%;"
        >
          CREATED BY
        </div>
        <div
          class="cell"
          colspan="1"
          role="columnheader"
          style="width: 15%;"
        >
          ACTIONS
        </div>
        <div
          class="cell"
          colspan="1"
          role="columnheader"
          style="width: 5%;"
        >
          
        </div>
      </div>
      <div
        class="row card clickable"
        role="row"
      >
        <div
          class="cell"
          role="cell"
          style="width: 25%;"
        >
          <div
            class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
          >
            <span
              class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-blue500"
              data-icon="step-group"
            >
              <svg
                height="30"
                id="step-group_svg__Layer_1"
                viewBox="0 0 21 20"
                width="30"
                x="0"
                xml:space="preserve"
                y="0"
              >
                <style>
                  .step-group_svg__st0{fill:#fff!important}
                </style>
                <path
                  class="step-group_svg__st0"
                  d="M1 .8h13v13.3H1z"
                />
                <path
                  d="M14 1H1v13h13V1zm1-1v15H0V0h15z"
                />
                <path
                  class="step-group_svg__st0"
                  d="M7 6h13v13H7z"
                />
                <path
                  class="step-group_svg__st1"
                  d="M20 6H7v13h13V6zm1-1v15H6V5h15z"
                />
              </svg>
            </span>
            <div>
              <div
                class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
                data-testid="OverLayInput"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black"
                >
                  OverLayInput
                </p>
              </div>
              <p
                class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400"
              >
                OverLayInput
              </p>
            </div>
          </div>
        </div>
        <div
          class="cell"
          role="cell"
          style="width: 20%;"
        >
          <p
            class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--color StyledProps--color-grey400"
            style="--text-line-clamp: 2;"
          >
            OverLayInput
          </p>
        </div>
        <div
          class="cell"
          role="cell"
          style="width: 10%;"
        >
          
        </div>
        <div
          class="cell"
          role="cell"
          style="width: 10%;"
        >
          
        </div>
        <div
          class="cell"
          role="cell"
          style="width: 10%;"
        >
          
        </div>
        <div
          class="cell"
          role="cell"
          style="width: 15%;"
        >
          <span
            class=""
          >
            <button
              class="bp3-button StyledProps--font Button--button StyledProps--main StyledProps--intent-primary"
              type="button"
            >
              <span
                class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall"
                data-icon="run-pipeline"
              >
                <svg
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <defs>
                    <path
                      d="M7.2 10.6V5.4L10.6 8l-3.4 2.6zm4.4-3.3L7.4 4.1c-.3-.2-.6-.2-.9-.1-.3.2-.5.5-.5.8v6.4c0 .3.2.6.5.8.3.1.6.1.9-.1l4.2-3.2c.2-.2.3-.4.3-.7s-.1-.5-.3-.7zm1.1 5.4c-2.2 2.2-5.7 2.6-8.4.8s-3.6-5.1-2.4-8c1.2-2.9 4.3-4.6 7.4-4 3.1.6 5.3 3.3 5.3 6.5.1 1.8-.6 3.4-1.9 4.7zm2.7-7.8C13.8 1.1 9.6-.9 5.7.4 1.7 1.6-.7 5.6.2 9.6s4.5 6.8 8.6 6.4c4.1-.4 7.2-3.8 7.2-8 0-1.1-.2-2.1-.6-3.1z"
                      id="run-pipeline_svg__a"
                    />
                  </defs>
                  <clippath
                    id="run-pipeline_svg__b"
                  >
                    <use
                      overflow="visible"
                      xlink:href="#run-pipeline_svg__a"
                    />
                  </clippath>
                  <path
                    clip-path="url(#run-pipeline_svg__b)"
                    d="M-5-5h26v26H-5z"
                    fill="#54b610"
                  />
                </svg>
              </span>
              <span
                class="bp3-button-text"
              >
                Run Pipeline
              </span>
            </button>
          </span>
        </div>
        <div
          class="cell"
          role="cell"
          style="width: 5%;"
        >
          <div
            class="Layout--horizontal StyledProps--main"
            style="justify-content: flex-end;"
          >
            <span
              class="bp3-popover-wrapper bp3-dark"
            >
              <span
                class="bp3-popover-target"
              >
                <button
                  class="bp3-button bp3-minimal StyledProps--font Button--button Button--iconOnly StyledProps--main actionButton"
                  type="button"
                >
                  <span
                    class="bp3-icon bp3-icon-more StyledProps--main"
                    icon="more"
                  >
                    <svg
                      data-icon="more"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                    >
                      <desc>
                        more
                      </desc>
                      <path
                        d="M2 6.03a2 2 0 100 4 2 2 0 100-4zM14 6.03a2 2 0 100 4 2 2 0 100-4zM8 6.03a2 2 0 100 4 2 2 0 100-4z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </div>
        </div>
      </div>
      <div
        class="row card clickable"
        role="row"
      >
        <div
          class="cell"
          role="cell"
          style="width: 25%;"
        >
          <div
            class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
          >
            <span
              class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-black"
              data-icon="yaml-builder-input-sets"
            >
              <svg
                height="30"
                viewBox="0 0 18 16"
                width="30"
              >
                <g
                  fill="#22272D"
                >
                  <path
                    d="M12.946 5.347a.49.49 0 00-.492-.486H4.581a.49.49 0 00-.492.486c0 .269.22.486.492.486h7.873a.49.49 0 00.492-.486zM4.581 8.264c-.272 0-.492.218-.492.486s.22.486.492.486H9.01c.271 0 .492-.218.492-.486s-.22-.486-.492-.486H4.58zm7.92 5.833H3.644a.55.55 0 01-.538-.56V2.018a.55.55 0 01.538-.56h9.748a.55.55 0 01.54.56v5.325c0 .269.22.486.491.486a.49.49 0 00.492-.486V2.02c0-.844-.679-1.533-1.523-1.533H3.643c-.844 0-1.523.69-1.523 1.533v11.517c0 .844.68 1.533 1.523 1.533h8.859a.49.49 0 00.492-.486.49.49 0 00-.492-.486z"
                  />
                  <path
                    d="M15.676 10.418h-1.005V9.413a.524.524 0 10-1.048 0v1.004H12.28a.524.524 0 000 1.049h1.343v1.343a.524.524 0 101.048 0v-1.343h1.005a.524.524 0 000-1.048z"
                  />
                </g>
              </svg>
            </span>
            <div>
              <div
                class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
                data-testid="asd"
              >
                <p
                  class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black"
                >
                  asd
                </p>
              </div>
              <p
                class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400"
              >
                asd
              </p>
            </div>
          </div>
        </div>
        <div
          class="cell"
          role="cell"
          style="width: 20%;"
        >
          <p
            class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--color StyledProps--color-grey400"
            style="--text-line-clamp: 2;"
          >
            asd
          </p>
        </div>
        <div
          class="cell"
          role="cell"
          style="width: 10%;"
        >
          
        </div>
        <div
          class="cell"
          role="cell"
          style="width: 10%;"
        >
          
        </div>
        <div
          class="cell"
          role="cell"
          style="width: 10%;"
        >
          
        </div>
        <div
          class="cell"
          role="cell"
          style="width: 15%;"
        >
          <span
            class=""
          >
            <button
              class="bp3-button StyledProps--font Button--button StyledProps--main StyledProps--intent-primary"
              type="button"
            >
              <span
                class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall"
                data-icon="run-pipeline"
              >
                <svg
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  <defs>
                    <path
                      d="M7.2 10.6V5.4L10.6 8l-3.4 2.6zm4.4-3.3L7.4 4.1c-.3-.2-.6-.2-.9-.1-.3.2-.5.5-.5.8v6.4c0 .3.2.6.5.8.3.1.6.1.9-.1l4.2-3.2c.2-.2.3-.4.3-.7s-.1-.5-.3-.7zm1.1 5.4c-2.2 2.2-5.7 2.6-8.4.8s-3.6-5.1-2.4-8c1.2-2.9 4.3-4.6 7.4-4 3.1.6 5.3 3.3 5.3 6.5.1 1.8-.6 3.4-1.9 4.7zm2.7-7.8C13.8 1.1 9.6-.9 5.7.4 1.7 1.6-.7 5.6.2 9.6s4.5 6.8 8.6 6.4c4.1-.4 7.2-3.8 7.2-8 0-1.1-.2-2.1-.6-3.1z"
                      id="run-pipeline_svg__a"
                    />
                  </defs>
                  <clippath
                    id="run-pipeline_svg__b"
                  >
                    <use
                      overflow="visible"
                      xlink:href="#run-pipeline_svg__a"
                    />
                  </clippath>
                  <path
                    clip-path="url(#run-pipeline_svg__b)"
                    d="M-5-5h26v26H-5z"
                    fill="#54b610"
                  />
                </svg>
              </span>
              <span
                class="bp3-button-text"
              >
                Run Pipeline
              </span>
            </button>
          </span>
        </div>
        <div
          class="cell"
          role="cell"
          style="width: 5%;"
        >
          <div
            class="Layout--horizontal StyledProps--main"
            style="justify-content: flex-end;"
          >
            <span
              class="bp3-popover-wrapper bp3-dark"
            >
              <span
                class="bp3-popover-target"
              >
                <button
                  class="bp3-button bp3-minimal StyledProps--font Button--button Button--iconOnly StyledProps--main actionButton"
                  type="button"
                >
                  <span
                    class="bp3-icon bp3-icon-more StyledProps--main"
                    icon="more"
                  >
                    <svg
                      data-icon="more"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                    >
                      <desc>
                        more
                      </desc>
                      <path
                        d="M2 6.03a2 2 0 100 4 2 2 0 100-4zM14 6.03a2 2 0 100 4 2 2 0 100-4zM8 6.03a2 2 0 100 4 2 2 0 100-4z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </div>
        </div>
      </div>
      <div
        class="Layout--horizontal StyledProps--main Pagination--container StyledProps--padding-top-medium StyledProps--padding-bottom-medium StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-distribution-space-between"
      >
        <span
          class="StyledProps--font StyledProps--main Pagination--text StyledProps--inline"
        >
          Showing 
          <strong>
            3
          </strong>
           of 
          <strong>
            3
          </strong>
        </span>
      </div>
    </div>
  </div>
</div>
`;
